### PR TITLE
Fix package.json repository.url; fix travis build

### LINKED ISSUE #2062: package.json repository.url format invalid
TL;DR: package.json repository.url should be `github:Glavin001/atom-beautify`

Repository URL in `package.json` should be publicly available:
https://docs.npmjs.com/files/package.json#repository

When viewing the package from the Atom package browser, you can't click the URL (it shows up as http://git@github.com/Glavin001/atom-beautify)

I guess it should be something like `github:Glavin001/atom-beautify`

(It's so minor I feel ashamed to open the issue, but here you go 😛)

### PR DESCRIPTION
**Pull request by joaomlneto**
### Changes

- Fix `repository` field in `package.json`
  - [Repository URL in `package.json` should be publicly available](https://docs.npmjs.com/files/package.json#repository)
  - When viewing the package from the package browser in Atom, URL is malformed -- shows up as http://git@github.com/Glavin001/atom-beautify

- Tentative fix for Travis Build
  - Removed `npm install -g coffee-formatter` from `before_install`

- `package.json`:
  - Added `npm run lint` to `pretest` script

### Issues Closed

#2062: package.json repository.url format invalid

### Checklist

Check all those that are applicable and complete.

- [x] Merged with latest `master` branch
- [x] Regenerate documentation with `npm run docs`
- [x] Add change details to `CHANGELOG.md` under "Next" section
- [x] ~~Added examples for testing to [examples/ directory](examples/)~~
- [x] Travis CI passes (Mac support)
- [x] AppVeyor passes (Windows support)

### CODE REVIEW

**Comment by Glavin001** [comment]
Looking at the npm docs: https://docs.npmjs.com/files/package.json#repository

image

I wonder if 

```
"repository": {
     "type": "git",
     "url": "github:Glavin001/atom-beautify"
   },
```

should really be

```
"repository": "github:Glavin001/atom-beautify",
```

?

**Comment by joaomlneto** [comment]
You're correct. Sorry for overlooking that!

**Comment by Glavin001** [comment]
Travis CI is failing for Linux builds after this change.

```
$ npm install coffee-formatter
npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning EINTEGRITY: sha512-yJHVQEhyqPLUTgt9B83PXu6W3rx4MvvHvSUvToogpwoGDOUQ+yDrR0HRot+yOCdCO7u4hX3pWft6kWBBcqh0UA== integrity checksum failed when using sha512: wanted sha512-yJHVQEhyqPLUTgt9B83PXu6W3rx4MvvHvSUvToogpwoGDOUQ+yDrR0HRot+yOCdCO7u4hX3pWft6kWBBcqh0UA== but got sha1-UWbihkV/AzBgZL5Ul+jbsMPTIIM=. (11423 bytes)
npm WARN registry Using stale package data from https://registry.npmjs.org/ due to a request error during revalidation.
npm WARN ajv-keywords@3.1.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm ERR! Cannot read property '0' of undefined
   ╭─────────────────────────────────────╮
   │                                     │
   │   Update available 5.5.1 → 5.7.1    │
   │     Run npm i -g npm to update      │
   │                                     │
   ╰─────────────────────────────────────╯
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-03-03T00_21_44_449Z-debug.log
The command "npm install coffee-formatter" failed and exited with 1 during .
```

It looks like `npm` version may be out-of-date on the Linux machines and not support this change properly.

See https://github.com/npm/npm/issues/17858#issuecomment-350189251

> Gonna close this, since it seems to be solved for people on 5.6.0

Let's force update `npm` to `5.7` by adding to `.travis.yml`: https://github.com/travis-ci/travis-ci/issues/4653#issuecomment-194051953

```
npm i -g npm@5.7
```

**Comment by szeck87** [comment]
It was doing that before this change.  I thought it was a cache issue as it went away up after clearing it, but I guess not.  Probably one of the dependency changes from 0.32.0.

**Comment by joaomlneto** [comment]
Still not passing.

```bash
9.40s$ npm install coffee-formatter
npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN ajv-keywords@3.1.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm ERR! Cannot read property '0' of undefined
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-03-03T20_21_25_600Z-debug.log
The command "npm install coffee-formatter" failed and exited with 1 during .
Your build has been stopped.
```

Couldn't we simply run `coffeelint` on `pretest` script instead of `coffee-formatter`?
It doesn't format the code -- simply checks it (but I guess that's what we want in this case!)

I noticed it's being used already in the package-json (script `lint`), and I'm happy using it in my projects.

> Let's force update `npm` to `5.7` by adding to `.travis.yml`

Should also add to `package.json`? I feel I'm being too pedantic without any real benefit :-)
https://docs.npmjs.com/files/package.json#engines

File: package.json [hunk 1]
```diff
@@ -456,6 +456,7 @@
     "prepare": "npm run docs",
     "lint": "coffeelint src/ spec/",
     "code-docs": "codo && open docs/code/index.html",
+    "pretest": "npm lint",
```
Comment (Glavin001) [inline package.json]: Should this be `npm run lint`?

Comment (Glavin001) [reply package.json]: Also note `npm test` (AKA `npm run test`) is not run for Travis CI. Instead it calls only `sh build-package.sh`.
See https://github.com/joaomlneto/atom-beautify/blob/a766cbc34b7a41d2c17071c4c52332b5e1b4adba/.travis.yml#L12 

Fortunately, `coffeelint` is already called by https://github.com/joaomlneto/atom-beautify/blob/a766cbc34b7a41d2c17071c4c52332b5e1b4adba/build-package.sh#L20-L31 so this is OK.

File: .travis.yml [hunk 2]
```diff
@@ -82,8 +82,6 @@ before_install:
   - composer install
   # NPM Version
   - npm i -g npm@5.7
-  # CoffeeScript
-  - npm install coffee-formatter
```
Comment (Glavin001) [inline .travis.yml]: I don't see where this is being used. Maybe old. If it passes tests then I suppose it is OK to remove 👍 .

**Comment by Glavin001** [comment]
@szeck87 : if you approve this as well, let's merge 👍 .

Thanks for contributing @joaomlneto!

**Comment by szeck87** [comment]
What's the reasoning behind installing npm 5.7 if coffee-formatter is being removed?

File: .travis.yml (hunk 2 above)
Comment (joaomlneto) [reply .travis.yml]: The tests pass, and I don't see it being used anywhere in the build process either.
It's not in `appveyor.yml`; but I find it weird that it passes in OSX and not in Linux

File: package.json (hunk 1 above)
Comment (joaomlneto) [reply package.json]: > Should this be npm run lint?

That's correct! 😨

**Comment by joaomlneto** [comment]
> What's the reasoning behind installing npm 5.7 if coffee-formatter is being removed?

I've reverted including the npm version in `travis.yml`.

**Comment by szeck87** [comment]
Ok thanks. Looking closer at Travis, I kind of wonder when/where node dependencies are installed. I don't see it in the logs at all. But the build succeeds with and without caching.

**Comment by Glavin001** [comment]
Good catch, @szeck87 !

Thank you for contributing @joaomlneto !

**Comment by Glavin001** [comment]
Wow. Looks like `apm` (Atom's package manager) does not support this!

```
❯ apm publish patch
Preparing and tagging a new version ✓
Pushing v0.32.1 tag ✗
ssh: Could not resolve hostname github: nodename nor servname provided, or not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by Glavin001** [comment]
Reverted as it broken `apm publish`. Will need to investigate later. 😞

**Comment by Glavin001** [comment]
Marked as published, since it was in v0.32.1, which failed. Will need to revisit this 😞

**Comment by joaomlneto** [comment]
I'll also take a look, but IMHO this should be fixed upstream as `apm` should support the same repository format as `npm`.
I found no existing issues on this over at [atom/apm](https://github.com/atom/apm)

**Comment by szeck87** [comment]
@joaomlneto which link doesn't work in the Atom settings/packages view?  I can't seem to find any issues anywhere.

**Comment by szeck87** [comment]
@joaomlneto I looked through other packages package.json file and they all just use https.  Can we just use that?

**Comment by joaomlneto** [comment]
> @joaomlneto I looked through other packages package.json file and they all just use https. Can we just use that?

HTTPS **should** work perfectly fine. That's how everyone is doing it, apparently.
I just found the `github:user/repo` shorthand existed for `npm` and extrapolated it should work for `apm`.
I didn't test it -- never had a go publishing an atom package

> @joaomlneto which link doesn't work in the Atom settings/packages view?

screen shot 2018-03-10 at 19 12 45
I know it's pretty minor! 🙁

**Comment by Glavin001** [comment]
> I'll also take a look, but IMHO this should be fixed upstream as apm should support the same repository format as npm.

I agree. If you create an Atom/apm issue, please link back here. I would consider adding this PR in again. 👍

**Comment by joaomlneto** [comment]
`apm` unofficially tagged it as `WONTFIX`.
I'll open a new PR with the `repository.url = https://` and travis fix.

**Pull request merged**